### PR TITLE
Clean DynELF TODO

### PR DESCRIPTION
**Pull request by hanhanW**
Clean DynELF TODO

* Add a little doc to `MemLeak`
* Use the constants in `elftools`.
* Move `find_base` from `MemLeak` to `DynELF`
* Use `ELF` instead of `objdump` in the test.

---
This change is [Reviewable](https://reviewable.io/reviews/peter50216/pwntools-ruby/35)

### CODE REVIEW

File: lib/pwnlib/dynelf.rb [hunk 1]
```diff
@@ -118,15 +128,17 @@ def find_dt(tag)
...
-      @strtab ||= find_dt(DT_STRTAB)
+      @strtab ||= find_dt(ELFTools::Constants::DT::DT_STRTAB)
     end
 
     def symtab
-      @symtab ||= find_dt(DT_SYMTAB)
+      @symtab ||= find_dt(ELFTools::Constants::DT::DT_SYMTAB)
     end
+
+    include Context
```
Comment (david942j) [inline lib/pwnlib/dynelf.rb]: not used?

**Comment by david942j** [comment]
why there's no travis Orz

**Comment by david942j** [comment]
LGTM
:lgtm:

---

Reviewed 7 of 12 files at r2, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/35#-:-KsS8gjin2wDpXDDulME:b3hosot)*

**Comment by hanhanW** [comment]
@david942j Dont be angry QQ

**Comment by david942j** [comment]
LGTM
= ^ w ^ =

**Comment by david942j** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/35)*

**Pull request merged**